packages/adapter-commons/src/service.ts

```diff
@@ -57,6 +57,10 @@ export abstract class AdapterBase<
    * @returns Wether or not multiple updates are allowed.
    */
   allowsMulti(method: string, params: ServiceParams = {} as ServiceParams) {
+    if (params.bulk) {
```

Comment:
Maybe we can update `allowMulti` to accept `context.id` and have a centralized place to throw an error whenever `params.bulk === true` and

- For create: `data` is an object
- For patch and remove: `context.id` is null